frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx

```diff
@@ -38,6 +40,35 @@ export function PolicyTemplateDetails(props: {
     const [relatedObjects, setRelatedObjects] = useState<any>()
     const [templateError, setTemplateError] = useState<string>()
     const [isExpanded, setIsExpanded] = useState<boolean>(true)
+    const [managedClusterAddOns] = useRecoilState(managedClusterAddonsState)
+
+    let templateClusterName = clusterName
+    let templateNamespace = clusterName
+
+    // Determine if the policy framework is deployed in hosted mode. If so, the policy template needs to be retrieved
+    // from the hosting cluster instead of the managed cluster.
+    for (const addon of managedClusterAddOns) {
+        if (addon.metadata.namespace !== clusterName) {
+            continue
+        }
+
+        if (addon.metadata.name !== 'governance-policy-framework') {
+            continue
+        }
+
+        if (addon.metadata.annotations?.['addon.open-cluster-management.io/hosting-cluster-name']) {
+            templateClusterName = addon.metadata.annotations['addon.open-cluster-management.io/hosting-cluster-name']
+
+            if (addon.spec.installNamespace) {
+                templateNamespace = addon.spec.installNamespace
+            } else {
+                // This is the default namespace but it shouldn't be used for hosted mode.
+                templateNamespace = 'open-cluster-management-agent-addon'
+            }
```

Comment:
Nit: is `templateNamespace = addon.spec.installNamespace || 'open-cluster-management-agent-addon'` equivalent? I don't know if it's really better, since the comment is more clear as-is.